### DISCUSSION TITLE
How to work a variable generated by a <script> file?

### DISCUSSION THREAD

**Discussion by rafael-legyo** [vercel/next.js#17922] [Help] [answered]
This variable receives several functions that I need to use on my page, along with my form (React Hook Form)

I added the script to Head
<script type="text/javascript" src="https://js.iugu.com/v2"></script>

But I can't use the variable

Example
`Iugu.setAccountID("COLOCAR SEU ACCOUNT ID AQUI");
`

**Comment by jamesmosier** [comment, marked as answer]
Hi @rafael-legyo. Do you happen to have an example of what isn't working?

If you go to your browser's console and type `Iugu` is anything returned?

**Reply by rafael-legyo** [reply]
Hi @jamesmosier 

Thanks for the feedback!

**I get:
ReferenceError: Iugu is not defined**

Here is an example from my code
`https://codesandbox.io/s/test-03znv?file=/pages/index.js:281-285`

Am I traveling, or should this work?

To me at the time I'm writing on CodeSandbox, it even appears:
![Screen Shot 2020-10-15 at 19 51 46](https://user-images.githubusercontent.com/48858913/96194122-14f67b80-0f20-11eb-9b88-3f596ec11ce6.png)

But localhost I get this same error:
![Screen Shot 2020-10-15 at 19 53 39](https://user-images.githubusercontent.com/48858913/96194191-3bb4b200-0f20-11eb-931e-b28282751633.png)

**Reply by jamesmosier** [reply]
The problem is that the script is not available when this page is server rendered. Therefore you have to wait for it to be ready:

```js
useEffect(() => {
  if (typeof Iugu !== 'undefined') {
    Iugu.setAccountID()
    // and any other methods for Iugu
  }
}, [])
```

**Reply by rafael-legyo** [reply]
Dude, I would never think of using useEffect for this. In fact, I only used it until today to make calls to api.

Will useEffect wait for the component (the page) to render and then run?

Thanks a lot for the help!!!

**Reply by jamesmosier** [reply]
> Will useEffect wait for the component (the page) to render and then run?

`useEffect` runs on first render of the component, so your script should be loaded by then although it is not guaranteed.
